[sailfish-browser] Removed remorse timer from delete bookmark. Contri…

antseppa opened the pull request:
…butes to JB#27571

rainemak commented on src/pages/components/FavoriteGrid.qml:
```diff
@@ -61,16 +61,8 @@ SilicaGridView {
 
-        property Item remorse
-
         function remove() {
```
I'd also remove this function and move ```bookmarkModel.removeBookmark(url)``` to the FavoriteContextMenu. Matter of taste as currently FavoriteContextMenu is not using ids from elsewhere and that's also good. Not big deal.

For safety, add url parameter for the remove function is you don't remove this. Now url refers to the model.url but it looks quite odd (maybe I have added at some point of time) as call is coming through FavoriteContextMenu. Url is passed to the context menu in the show.

antseppa commented:
Updated: removed remove() function and moved bookmarkModel.removeBookmark(url) to context menu.

antseppa commented:
Updated: added a commit which updates indices in bookmarkIndexes map after a bookmark has been removed. Without this, consecutive calls to DeclarativeBookmarkModel::removeBookmark either crashed browser or caused wrong bookmark to be deleted if the index of the bookmark was greater than previous one.

rainemak commented on src/pages/components/FavoriteContextMenu.qml:
```diff
@@ -50,7 +50,9 @@ Component {
-            onClicked: delegate.remove()
+            onClicked: {
+                bookmarkModel.removeBookmark(url)
+            }
```
No need for brackets.

rainemak commented:
LGTM
